internal/app/connectconformance/test_case_library.go

```diff
@@ -566,31 +566,43 @@ func populateExpectedUnaryResponse(testCase *conformancev1.TestCase) error {
 		ResponseTrailers: def.ResponseTrailers,
 	}
 
-	switch respType := def.Response.(type) {
-	case *conformancev1.UnaryResponseDefinition_Error:
-		// If an error was specified, it should be returned in the response
-		expected.Error = respType.Error
-
-		// Unary responses that return an error should have the request info
-		// in the error details
-		reqInfoAny, err := anypb.New(reqInfo)
-		if err != nil {
-			return connect.NewError(connect.CodeInternal, err)
-		}
-		respType.Error.Details = append(respType.Error.Details, reqInfoAny)
-	case *conformancev1.UnaryResponseDefinition_ResponseData, nil:
-		// If response data was specified for the response (or nothing at all),
-		// the server should echo back the request message and headers in the response
-		payload := &conformancev1.ConformancePayload{
-			RequestInfo: reqInfo,
+	if def.RawResponse != nil {
+		// TODO - Handle other raw response values
+		// If an HTTP error status was specified in the raw response, then the test case is
+		// forcing the return of an HTTP error, which should be mapped to a Connect error
+		// according to the protocol.
+		if def.RawResponse.StatusCode >= 400 && def.RawResponse.StatusCode <= 599 {
+			expected.Error = &conformancev1.Error{
+				Code: mapHTTPtoConnectCode(def.RawResponse.StatusCode),
+			}
```

Comment:
Not quite sure about this whole block. I know we discussed that you could specify a raw response _and_ the standard response definition fields, but I was curious about two things:

- The [proto comment](https://github.com/connectrpc/conformance/blob/main/proto/connectrpc/conformance/v1/service.proto#L168) mentions it should be one or the other in test cases
- If the comment is incorrect, then how should we best handle the case where an HTTP error code is specified in the raw response along with a defined `Error` to be returned? Should the defined `Error` override the raw response error in `expected`?

Reply:
I don't have a strong opinion here. I can see, for clarity, it would be best to just disallow the standard response definition.

But I also think, for consistency, there are things we'll want to tweak in how raw _requests_ work (since it is currently _necessary_ to specify both there).

Having said all that, I don't think we want this block. For raw responses, we should just require that the expected response be specified in the YAML file. That makes the test cases much more clear.